README.rst

```diff
@@ -26,21 +26,18 @@ First, run the server::
 
   $ stack run
 
-Then configure Stripe with a `webhook`_ pointing at the server and receiving the *charge.successful* event.
-Configure Stripe with ``http://<youraddress>:8081/v1/stripe/webhook``.
+Then create a testing charge::
 
-Then create a testing charge using Stripe::
-
-   $ STRIPE_SECRET_KEY=sk_test_...
    $ curl \
-     https://api.stripe.com/v1/charges   \
-     -u ${STRIPE_SECRET_KEY}:   \
-     -d amount=999   \
-     -d currency=usd   \
-     -d source=tok_visa   \
-     -d 'metadata[Voucher]=abcdefghijk'
+     http://<youraddress>:8081/v1/stripe/charge \
```

Comment:
I don't think this workflow will produce a good outcome for anyone trying to follow it.  The server will call out to Stripe to verify the token supplied, Stripe will respond that it is bogus, and the voucher will not be marked as paid.

If the developer is running a dummy Stripe server and has configured PaymentServer to point at it instead of the real Stripe endpoint, it could work.  PaymentServer doesn't come with a dummy Stripe server - but we have one in the PrivateStorageio system tests.  Possibly we could move it here and encourage developers to run and use it, thus enabling this kind of simple manual testing workflow.

The webhook stuff is even *more* wrong though so I don't mind this change - but we will need to go even further if we actually want to tell folks how they can successfully poke at PaymentServer at this level.

Reply:
If the server is configured with a test key, it *will* work, and you can get a test key from https://stripe.com/docs/development/quickstart (without an account even). I've not copied-pasted these instructions, but I wrote them based on a command I successfully ran against a test environment with such a key.

Reply:
Huh I didn't realize Stripe had *public* test-mode keys.  And I'm confused about their behavior.  Under real usage, the browser first creates the charge by invoking a Stripe API and then PaymentServer confirms/finalizes the charge using a different Stripe API and a token supplied by the browser.

But it sounds like with this test key, even without the first Stripe API call to create the charge and an associated token, the second Stripe API call with the "tok_visa" token succeeds?  Maybe because "tok_visa" is another magic value that the Stripe API respects when using the test key...?  I tried to find docs supporting this conclusion but the Stripe API docs search function doesn't seem very good and I didn't try very hard.

Reply:
> Under real usage, the browser first creates the charge by invoking a Stripe API and then PaymentServer confirms/finalizes the charge using a different Stripe API and a token supplied by the browser.

I think this isn't entirely accurate. I think the browser connects to token, but that token just represents the card/customer details, and *not* a charge. It then passes that token to our server,  which creates the charge. I found mention of the test tokens on https://stripe.com/docs/testing#cards. under the tokens tab (it took me a couple of times looking at that page to discover it). So, it seems `tok_visa` represents an arbitrary verified card, for us to create the charge with.